eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkService.java

```diff
@@ -16,62 +19,89 @@ public class AfkService {
     private final UserManager userManager;
 
     private final Map<UUID, Afk> afkByPlayer = new HashMap<>();
-    private final Map<UUID, Integer> interactions = new HashMap<>();
+    private final Map<UUID, Integer> interactionsCount = new HashMap<>();
+    private final Map<UUID, Instant> lastInteraction = new HashMap<>();
 
     public AfkService(AfkSettings afkSettings, NoticeService noticeService, UserManager userManager) {
         this.afkSettings = afkSettings;
         this.noticeService = noticeService;
         this.userManager = userManager;
     }
 
+    public void switchAfk(UUID player, AfkReason reason) {
+        if (this.isAfk(player)) {
+            this.clearAfk(player);
+            return;
+        }
+
+        this.markAfk(player, reason);
+    }
+
     public Afk markAfk(UUID player, AfkReason reason) {
         Afk afk = new Afk(player, reason, Instant.now());
 
         this.afkByPlayer.put(player, afk);
-        this.notifyAfk(player, false);
+        this.sendAfkNotification(player, true);
+
         return afk;
     }
 
-    public Afk markAfk(UUID player) {
-        return this.markAfk(player, AfkReason.PLUGIN);
+    public void clearAfk(UUID player) {
+        Afk afk = this.afkByPlayer.remove(player);
+
+        if (afk == null) {
+            return;
+        }
+
+        this.interactionsCount.remove(player);
+        this.lastInteraction.remove(player);
+        this.sendAfkNotification(player, false);
     }
 
     public boolean isAfk(UUID player) {
         return this.afkByPlayer.containsKey(player);
     }
 
     public void markInteraction(UUID player) {
-        int interactions = this.interactions.getOrDefault(player, 0);
+        this.lastInteraction.put(player, Instant.now());
 
-        interactions++;
+        int count = this.interactionsCount.getOrDefault(player, 0);
+        count++;
 
-        if (interactions >= this.afkSettings.interactionsCountDisableAfk()) {
+        if (count >= this.afkSettings.interactionsCountDisableAfk()) {
             this.clearAfk(player);
             return;
         }
 
-        this.interactions.put(player, interactions);
+        this.interactionsCount.put(player, count);
     }
 
-    public boolean clearAfk(UUID player) {
-        Afk afk = this.afkByPlayer.remove(player);
+    public void checkLastMovement() {
+        Duration afkTimeout = this.afkSettings.getAfkInactivityTime();
+        Instant now = Instant.now();
 
-        if (afk == null) {
-            return false;
-        }
+        for (Player player : Bukkit.getOnlinePlayers()) {
```

Comment:
```suggestion
        for (Player player : Bukkit.getOnlinePlayers()) {
```

Tutaj wypadałoby użyć `#getServer#getOnlinePlayers`